containers/datalab/content/run-with-gce.sh

```diff
@@ -0,0 +1,145 @@
+ERR_LOGIN=2
+ERR_PROJECT_NOT_FOUND=3
+ERR_ZONE_NOT_FOUND=4
+ERR_INSTANCE_NOT_FOUND=5
+ERR_DEPLOY=6
+ERR_TUNNEL_FAILED=7
+ERR_GATEWAY_FAILED=7
+
+# Run the 'setup-env' script to ensure that gcloud has been told to use
+# a config directory under the mounted volume (so that the results of
+# 'gcloud auth login' are persisted).
+source /datalab/setup-env.sh
+export HOME=/content
+
+export PROJECT_ID=${PROJECT_ID:-`gcloud config list -q --format 'value(core.project)' 2> /dev/null`}
+export ZONE=${ZONE:-`gcloud config list -q --format 'value(compute.zone)' 2> /dev/null`}
+
+if [[ -n "${GATEWAY_VM}" ]]; then
+  GATEWAY_PART_1=`echo "${GATEWAY_VM}" | cut -d '/' -f 1`
+  GATEWAY_PART_2=`echo "${GATEWAY_VM}" | cut -d '/' -f 2`
+  GATEWAY_PART_3=`echo "${GATEWAY_VM}" | cut -d '/' -f 3`
+
+  if [[ -z "${GATEWAY_PART_3}" &&  -z "${GATEWAY_PART_3}" ]]; then
+    export INSTANCE="${GATEWAY_PART_1}"
+  elif [[ -z "${GATEWAY_PART_1}" || -z "${GATEWAY_PART_2}" || -z "${GATEWAY_PART_3}" ]]; then
+    echo "Malformed gateway VM name"
+    echo "${USAGE}"
+    exit "${ERR_MALFORMED_GATEWAY}"
+  else
+    export PROJECT_ID="${GATEWAY_PART_1}"
+    export ZONE="${GATEWAY_PART_2}"
+    export INSTANCE="${GATEWAY_PART_3}"
+  fi
+fi
+
+if [[ -n "${INSTANCE}" ]]; then
+  USER_EMAIL=`gcloud info --format="value(config.account)"`
+  if [[ -z "${USER_EMAIL}" ]]; then
+    FAILED_LOGIN=""
+    gcloud auth login || FAILED_LOGIN="true"
+    if [[ -n "${FAILED_LOGIN}" ]]; then
+      echo "Failed to log in to gcloud"
+      exit "${ERR_LOGIN}"
+    fi
+  fi
+
+  PROJECT_NOT_FOUND=""
+  ZONE_NOT_FOUND=""
+  INSTANCE_NOT_FOUND=""
+
+  if [[ -z "${PROJECT_ID}" ]]; then
+    read -p "Please enter the Google Cloud Platform project to use: " PROJECT_ID
+  fi
+  # Verify that the specified project exists...
+  gcloud -q projects describe "${PROJECT_ID}" >/dev/null 2>&1 || PROJECT_NOT_FOUND="true"
+  if [[ -n "${PROJECT_NOT_FOUND}" ]]; then
+    echo "Project ${PROJECT_ID} not found"
+    echo "${USAGE}"
+    exit "${ERR_PROJECT_NOT_FOUND}"
+  fi
+  # Persist the project for future runs.
+  gcloud config set project "${PROJECT_ID}"
+
+  if [[ -z "${ZONE}" ]]; then
+    read -p "Please enter the zone where the VM should be located: " ZONE
+  fi
+  # Verify that the specified zone exists...
+  gcloud -q compute zones describe "${ZONE}" >/dev/null 2>&1 || ZONE_NOT_FOUND="true"
+  if [[ -n "${ZONE_NOT_FOUND}" ]]; then
+    echo "Zone ${ZONE} not found"
+    echo "${USAGE}"
+    exit "${ERR_ZONE_NOT_FOUND}"
+  fi
+  # Persist the project and zone for future runs.
+  gcloud config set compute/zone "${ZONE}"
+
+  # Verify that the specified instance exists...
+  gcloud -q compute instances describe "${INSTANCE}" >/dev/null 2>&1 || INSTANCE_NOT_FOUND="true"
+  if [[ -n "${INSTANCE_NOT_FOUND}" ]]; then
+    echo "Instance ${INSTANCE} not found"
+    if [[ "${DEPLOY_VM}" == "true" ]]; then
+      /datalab/deploy.sh "${PROJECT_ID}" "${ZONE}" "${INSTANCE}" || exit ${ERR_DEPLOY}
```

Comment:
Ah. Sorry! I saw the renaming now, which I overlooked since there is no code change on that one.